src/engine/source/logicexpr/interface/logicexpr/logicexpr.hpp

```diff
@@ -0,0 +1,87 @@
+#ifndef _LOGIC_EXPRESSION_H
+#define _LOGIC_EXPRESSION_H
+
+#include <functional>
+#include <memory>
+#include <stdexcept>
+#include <string>
+
+#include <fmt/format.h>
+
+#include "evaluator.hpp"
+#include "parser.hpp"
+
+namespace logicexpr
+{
+
+/**
+ * @brief Generate evaluation function from a string logic expression.
+ * This function parses the string and generates a token tree, then uses the
+ * provided builder to generate the expression tree with all term's functions.
+ * Finally generates the function from built expression tree.
+ *
+ * @tparam Event Type of the event to be evaluated.
+ * @param expression String logic expression.
+ * @param termBuilder Builder to generate the term's evaluation function from
+ * its description.
+ * @param termParser Parser to parse the term's of the expression.
+ * @return std::function<bool(Event)> Evaluation function.
+ */
+template<typename Event, typename TermType, typename TermBuilder, typename TermParser>
+std::function<bool(Event)>
+buildDijstraEvaluator(const std::string& expression, TermBuilder&& termBuilder, TermParser&& termParser)
```

Comment:
nitpick - desing: Here the only thing that is treated differently is when it is a term, otherwise you could use this to avoid repeated code: 
```
        const auto& token = tokenExpr->m_token;

        if (token->isNot() || token->isOr() || token->isAnd())
        {
            builtExpr.m_type = token->isNot() ? evaluator::ExpressionType::NOT :
                                token->isOr() ? evaluator::ExpressionType::OR : evaluator::ExpressionType::AND;
            
            builtExpr.m_left = visitRef(tokenExpr->m_left, visitRef);

            if (builtExpr.m_type != evaluator::ExpressionType::NOT)
                builtExpr.m_right = visitRef(tokenExpr->m_right, visitRef);

            return builtExpr;
        }
```